### DISCUSSION TITLE
Jupyter: Same frame for images and videos

### DISCUSSION THREAD

**Discussion by kolibril13** [ManimCommunity/manim#2787] [Suggestions and Proposals]
Currently, videos in jupyter can be scaled with e.g. `config.media_width= "40.0vw"` and they have a nice list to download the video and to display it full screen.

![image](https://user-images.githubusercontent.com/44469195/113488434-6985b700-94be-11eb-8ba4-f41b65a39b75.png)

Unfortunately, images don't have any of these options (scaling, fullscreen, download) yet:

![image](https://user-images.githubusercontent.com/44469195/113488974-b323d100-94c1-11eb-8f36-31ecea2713e2.png)

Can we maybe embed images as well into the HTML video player and hide the "play" button?
That would be amazing!

**Comment by behackl** [comment]
I'm not aware of any simple out-of-the-box supported solution, we would likely need to include some sort of image gallery library.

**Comment by kolibril13** [comment]
Some googling got me these ideas:
* https://github.com/maartenbreddels/ipywebrtc
* https://stackoverflow.com/questions/18545077/image-fullscreen-on-click
* https://github.com/scottlittle/expand-cell-fullscreen

**Comment by ianhi** [comment]
Are you looking for something like the toolbar that [ipympl](https://github.com/matplotlib/ipympl#ipympl) (interactive in notebook matplotlib) has?
![image](https://user-images.githubusercontent.com/10111092/113492910-253bec00-94a9-11eb-961b-ff8640e45b4e.png)

You can construct this out of ipywidgets buttons and `display` it to show it.

**Comment by kolibril13** [comment]
Thanks @ianhi !
Here is an implementation, how it could look like in manim, it only needs a bit of adjustment for the current jupyter notebook structure.
@behackl, @Darylgolden @jsonvillanueva  : What do you think of this?

```python
from ipywidgets import GridspecLayout, AppLayout, Button, Layout, widgets
from PIL import Image
import webbrowser

file = open("new_pi.png", "rb")

image = file.read()
dis_img = widgets.Image(value=image, format="png")
dis_but = widgets.ToggleButton(
    value=False,
    description="Fullscreen",
    disabled=False,
    button_style="",
    tooltip="Description",
)

b1 = Button(
    description="",
    icon="fa-expand",
    button_style="",
    layout=Layout(height="30px", width="40px"),
)
b2 = Button(
    description="",
    icon="fa-download",
    button_style="",
    layout=Layout(height="30px", width="40px"),
)

button_list = GridspecLayout(2, 1, height="80px")
button_list[0, 0] = b1
button_list[1, 0] = b2

dis_img.width = "100px"


def on_button_expand_clicked(b):
    if dis_img.width != "100px":
        dis_img.width = "100px"
    else:
        dis_img.width = "300px"


def on_button_download_clicked(b):
    im1 = Image.open("new_pi.png")
    im1 = im1.save("/tmp/download.png")
    url = "file:///tmp/download.png"
    webbrowser.open(url)


b1.on_click(on_button_expand_clicked)
b2.on_click(on_button_download_clicked)

AppLayout(left_sidebar=button_list, center=dis_img, pane_widths=["50px", 1, 0])
```

https://user-images.githubusercontent.com/44469195/114272548-e7921280-9a16-11eb-9b54-922f441a9c2c.mov
